Description: Configure transpilation as per different versions and transpilers
Forwarded: not-needed
Reviewed-By: Yadd <yadd@debian.org>
Last-Update: 2021-12-16

--- a/rollup.config.js
+++ b/rollup.config.js
@@ -1,11 +1,37 @@
-import buble from 'rollup-plugin-buble';
+var plugins = [];
+
+switch(process.env.TRANSPILE) {
+case 'babel6':
+    var babel=require('@rollup/plugin-babel');
+    const plugins_babel6 = [
+      babel({
+	babelrc: false,
+	presets: [[process.env.BABEL_PRESET, { modules: false }]],
+    }),
+    ];
+    plugins = plugins_babel6; break;
+case 'babel7':
+    var babel=require('@rollup/plugin-babel');
+    const plugins_babel7 = [
+      babel(),
+    ];
+    plugins = plugins_babel7; break;
+case 'none':
+    plugins = []; break;
+default:
+    var buble=require('@rollup/plugin-buble');
+    const plugins_buble = [
+    buble()
+    ];
+    plugins = plugins_buble;
+}
 
 var pkg = require('./package.json');
 
 export default {
 	input: 'index.js',
-	plugins: [buble()],
-        external: ['rollup-pluginutils'],
+	plugins: plugins,
+	external: ['@rollup/pluginutils'],
 	output: [
 		{
 			format: 'cjs',
